tensorflow/core/kernels/unsorted_segment_join_op.cc

```diff
@@ -156,6 +157,7 @@ class UnsortedSegmentJoinOp : public OpKernel {
 
  private:
   string separator_;
+  static constexpr int MAX_SEGMENTS = 100000; // subject to change
```

Comment:
Hi, thank you for the change! Would you mind clarifying where we got this upper bound from and why it is needed?